KarhooSDK/Network/HttpClient/TokenRefreshingHttpClient/Interactor/KarhooRefreshTokenInteractor.swift

```diff
@@ -28,15 +28,32 @@ final class KarhooRefreshTokenInteractor: RefreshTokenInteractor {
     }
 
     func refreshToken(completion: @escaping (Result<Bool>) -> Void) {
-        self.callback = completion
+        callback = completion
 
         guard tokenNeedsRefreshing() else {
             completion(Result.success(result: false))
             return
         }
 
-        guard let refreshToken = dataStore.getCurrentCredentials()?.refreshToken else {
-            completion(Result.failure(error: RefreshTokenError.noRefreshToken))
+        guard let refreshToken = dataStore.getCurrentCredentials()?.refreshToken, refreshToken.isEmpty == false else {
+            Karhoo.configuration.requestNewAuthenticationCredentials { [weak self] newCredentials in
+                guard let self = self else {
+                    completion(.failure(error: RefreshTokenError.memoryAllocationError))
+                    return
+                }
+
+                if let newCredentials = newCredentials {
```

Comment:
Let's get the updated credentials from the data store here instead of waiting for them from the DP